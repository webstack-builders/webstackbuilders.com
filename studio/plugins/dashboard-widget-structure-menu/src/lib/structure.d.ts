export declare function isSubscribable(thing: any): any;
export declare function isStructure(structure: any): any;
export declare function serializeStructure(item: any, context: any, resolverArgs?: never[]): any;
export declare function getDefaultStructure(): any;
export declare function loadStructure(): import("rxjs").Observable<unknown>;
//# sourceMappingURL=structure.d.ts.map