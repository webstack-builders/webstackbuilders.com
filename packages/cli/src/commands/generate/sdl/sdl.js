import path from 'path'

import boxen from 'boxen'
import camelcase from 'camelcase'
import chalk from 'chalk'
import Listr from 'listr'
import pascalcase from 'pascalcase'
import pluralize from 'pluralize'
import terminalLink from 'terminal-link'

import { getConfig } from '@redwoodjs/internal'

import {
  generateTemplate,
  transformTSToJS,
  getSchema,
  getPaths,
  writeFilesTask,
  getEnum,
} from '../../../lib'
import c from '../../../lib/colors'
import { yargsDefaults } from '../../generate'
import { ensureUniquePlural, relationsForModel } from '../helpers'
import { files as serviceFiles } from '../service/service'

const IGNORE_FIELDS_FOR_INPUT = ['id', 'createdAt', 'updatedAt']

const missingIdConsoleMessage = () => {
  const line1 =
    chalk.bold.yellow('WARNING') +
    ': Cannot generate CRUD SDL without an `@id` database column.'
  const line2 = 'If you are trying to generate for a many-to-many join table '
  const line3 = "you'll need to update your schema definition to include"
  const line4 = 'an `@id` column. Read more here: '
  const line5 = chalk.underline.blue(
    'https://redwoodjs.com/docs/schema-relations'
  )

  console.error(
    boxen(line1 + '\n\n' + line2 + '\n' + line3 + '\n' + line4 + '\n' + line5, {
      padding: 1,
      margin: { top: 1, bottom: 3, right: 1, left: 2 },
      borderStyle: 'single',
    })
  )
}

const modelFieldToSDL = (field, required = true, types = {}) => {
  if (Object.entries(types).length) {
    field.type =
      field.kind === 'object' ? idType(types[field.type]) : field.type
  }

  const dictionary = {
    Json: 'JSON',
    Decimal: 'Float',
  }

  return `${field.name}: ${field.isList ? '[' : ''}${
    dictionary[field.type] || field.type
  }${field.isList ? ']' : ''}${
    (field.isRequired && required) | field.isList ? '!' : ''
  }`
}

const querySDL = (model) => {
  return model.fields.map((field) => modelFieldToSDL(field))
}

const inputSDL = (model, required, types = {}) => {
  return model.fields
    .filter((field) => {
      return (
        IGNORE_FIELDS_FOR_INPUT.indexOf(field.name) === -1 &&
        field.kind !== 'object'
      )
    })
    .map((field) => modelFieldToSDL(field, required, types))
}

// creates the CreateInput type (all fields are required)
const createInputSDL = (model, types = {}) => {
  return inputSDL(model, true, types)
}

// creates the UpdateInput type (not all fields are required)
const updateInputSDL = (model, types = {}) => {
  return inputSDL(model, false, types)
}

const idType = (model, crud) => {
  if (!crud) {
    return undefined
  }

  const idField = model.fields.find((field) => field.isId)
  if (!idField) {
    missingIdConsoleMessage()
    throw new Error('Failed: Could not generate SDL')
  }
  return idField.type
}

const sdlFromSchemaModel = async (name, crud) => {
  const model = await getSchema(name)

  if (model) {
    // get models for user-defined types referenced
    const types = (
      await Promise.all(
        model.fields
          .filter((field) => field.kind === 'object')
          .map(async (field) => {
            const model = await getSchema(field.type)
            return model
          })
      )
    ).reduce((acc, cur) => ({ ...acc, [cur.name]: cur }), {})

    // Get enum definition and fields from user-defined types
    const enums = (
      await Promise.all(
        model.fields
          .filter((field) => field.kind === 'enum')
          .map(async (field) => {
            const enumDef = await getEnum(field.type)
            return enumDef
          })
      )
    ).reduce((acc, curr) => acc.concat(curr), [])

    return {
      query: querySDL(model).join('\n    '),
      createInput: createInputSDL(model, types).join('\n    '),
      updateInput: updateInputSDL(model, types).join('\n    '),
      idType: idType(model, crud),
      relations: relationsForModel(model),
      enums,
    }
  } else {
    throw new Error(
      `"${name}" model not found, check if it exists in "./api/prisma/schema.prisma"`
    )
  }
}

export const files = async ({ name, crud, tests, typescript }) => {
  const { query, createInput, updateInput, idType, relations, enums } =
    await sdlFromSchemaModel(pascalcase(pluralize.singular(name)), crud)

  let template = generateTemplate(
    path.join('sdl', 'templates', `sdl.ts.template`),
    {
      name,
      crud,
      query,
      createInput,
      updateInput,
      idType,
      enums,
    }
  )

  const extension = typescript ? 'ts' : 'js'
  let outputPath = path.join(
    getPaths().api.graphql,
    `${camelcase(pluralize(name))}.sdl.${extension}`
  )

  if (typescript) {
    template = transformTSToJS(outputPath, template)
  }

  return {
    [outputPath]: template,
    ...(await serviceFiles({ name, crud, tests, relations, typescript })),
  }
}

export const defaults = {
  ...yargsDefaults,
  crud: {
    default: false,
    description: 'Also generate mutations',
    type: 'boolean',
  },
}

export const command = 'sdl <model>'
export const description =
  'Generate a GraphQL schema and service component based on a given DB schema Model'
export const builder = (yargs) => {
  yargs
    .positional('model', {
      description: 'Model to generate the sdl for',
      type: 'string',
    })
    .option('tests', {
      description: 'Generate test files',
      type: 'boolean',
    })
    .epilogue(
      `Also see the ${terminalLink(
        'Redwood CLI Reference',
        'https://redwoodjs.com/reference/command-line-interface#generate-sdl'
      )}`
    )

  // Merge default options in
  Object.entries(defaults).forEach(([option, config]) => {
    yargs.option(option, config)
  })
}
// TODO: Add --dry-run command
export const handler = async ({ model, crud, force, tests, typescript }) => {
  if (tests === undefined) {
    tests = getConfig().generate.tests
  }
  const tasks = new Listr(
    [
      {
        title: 'Generating SDL files...',
        task: async () => {
          const f = await files({ name: model, tests, crud, typescript })
          return writeFilesTask(f, { overwriteExisting: force })
        },
      },
    ].filter(Boolean),
    { collapse: false, exitOnError: true }
  )

  try {
    await ensureUniquePlural({ model })
    await tasks.run()
  } catch (e) {
    console.error(c.error(e.message))
    process.exit(e?.exitCode || 1)
  }
}
