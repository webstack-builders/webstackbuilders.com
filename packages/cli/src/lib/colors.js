import chalk from 'chalk'

/**
 * To keep a consistent color/style palette between cli packages, such as
 * @redwood/cli and @redwood/create-redwood-app, please keep them compatible
 * with one and another. We'll might split up and refactor these into a
 * separate package when there is a strong motivation behind it.
 *
 * Current files:
 *
 * - packages/cli/src/lib/colors.js (this file)
 * - packages/create-redwood-app/src/create-redwood-app.js
 */
export default {
  error: chalk.bold.red,
  warning: chalk.keyword('orange'),
  green: chalk.green,
  info: chalk.grey,
  bold: chalk.bold,
  underline: chalk.underline,
}
