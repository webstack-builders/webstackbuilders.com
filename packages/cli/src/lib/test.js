/* eslint-env jest */

// Include at the top of your tests. Automatically mocks out the file system
//
// import { loadComponentFixture } from 'src/lib/test'
//
// test('true is true', () => {
//   expect('some output').toEqual(loadComponentFixture('component', 'filename.js'))
// })

import fs from 'fs'
import path from 'path'

jest.mock('@redwoodjs/internal', () => {
  const path = require('path')
  return {
    ...jest.requireActual('@redwoodjs/internal'),
    generate: () => {},
    getPaths: () => {
      const BASE_PATH = '/path/to/project'
      return {
        base: BASE_PATH,
        api: {
          dataMigrations: path.join(BASE_PATH, './api/prisma/dataMigrations'),
          db: path.join(global.__dirname, 'fixtures'), // this folder
          src: path.join(BASE_PATH, './api/src'),
          services: path.join(BASE_PATH, './api/src/services'),
          graphql: path.join(BASE_PATH, './api/src/graphql'),
          functions: path.join(BASE_PATH, './api/src/functions'),
        },
        web: {
          src: path.join(BASE_PATH, './web/src'),
          routes: path.join(BASE_PATH, 'web/src/Routes.js'),
          components: path.join(BASE_PATH, '/web/src/components'),
          layouts: path.join(BASE_PATH, '/web/src/layouts'),
          pages: path.join(BASE_PATH, '/web/src/pages'),
        },
        scripts: path.join(BASE_PATH, 'scripts'),
        generated: {
          base: path.join(BASE_PATH, '.redwood'),
          schema: path.join(BASE_PATH, '.redwood/schema.graphql'),
          types: {
            includes: path.join(BASE_PATH, '.redwood/types/includes'),
            mirror: path.join(BASE_PATH, '.redwood/types/mirror'),
          },
        },
      }
    },
  }
})

global.__prettierPath = path.resolve(
  __dirname,
  './__tests__/fixtures/prettier.config.js'
)

jest.mock('path', () => {
  const path = jest.requireActual('path')
  return {
    ...path,
    join: (...paths) => {
      if (
        paths &&
        paths[0] === '/path/to/project' &&
        paths[1] === 'prettier.config.js'
      ) {
        return global.__prettierPath
      }
      return path.join(...paths)
    },
  }
})

export const generatorsRootPath = path.join(
  __dirname,
  '..',
  'commands',
  'generate'
)

/**
 * Loads the fixture for a generator by assuming a lot of the path structure
 * automatically:
 *
 *   `loadGeneratorFixture('scaffold', 'NamePage.js')`
 *
 * will return the contents of:
 *
 *   `cli/src/commands/generate/scaffold/__tests__/fixtures/NamePage.js`
 */
export const loadGeneratorFixture = (generator, name) => {
  return loadFixture(
    path.join(
      __dirname,
      '..',
      'commands',
      'generate',
      generator,
      '__tests__',
      'fixtures',
      name
    )
  )
}

/**
 * Returns the contents of a text file in a `fixtures` directory
 */
export const loadFixture = (filepath) => {
  return fs.readFileSync(filepath).toString()
}
