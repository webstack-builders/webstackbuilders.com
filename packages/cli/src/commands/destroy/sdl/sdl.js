import Listr from 'listr'

import { ensureUniquePlural } from '../../../commands/generate/helpers'
import { deleteFilesTask } from '../../../lib'
import c from '../../../lib/colors'
import { files } from '../../generate/sdl/sdl'

export const command = 'sdl <model>'
export const description =
  'Destroy a GraphQL schema and service component based on a given DB schema Model'

export const builder = (yargs) => {
  yargs.positional('model', {
    description: 'Model to destroy the sdl of',
    type: 'string',
  })
}

export const tasks = ({ model }) =>
  new Listr(
    [
      {
        title: 'Destroying GraphQL schema and service component files...',
        task: async () => {
          const f = await files({ name: model })
          return deleteFilesTask(f)
        },
      },
    ],
    { collapse: false, exitOnError: true }
  )

export const handler = async ({ model }) => {
  await ensureUniquePlural({ model, inDestroyer: true })
  const t = tasks({ model })

  try {
    await t.run()
  } catch (e) {
    console.log(c.error(e.message))
  }
}
