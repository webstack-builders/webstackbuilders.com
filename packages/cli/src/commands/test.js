import execa from 'execa'
import terminalLink from 'terminal-link'

import { ensurePosixPath } from '@redwoodjs/internal'
import { getProject } from '@redwoodjs/structure'

import { getPaths } from '../lib'

// https://github.com/facebook/create-react-app/blob/cbad256a4aacfc3084be7ccf91aad87899c63564/packages/react-scripts/scripts/test.js#L39
function isInGitRepository() {
  try {
    execa.commandSync('git rev-parse --is-inside-work-tree')
    return true
  } catch (e) {
    return false
  }
}

function isInMercurialRepository() {
  try {
    execa.commandSync('hg --cwd . root')
    return true
  } catch (e) {
    return false
  }
}

export const command = 'test [filter..]'
export const description = 'Run Jest tests. Defaults to watch mode'
export const builder = (yargs) => {
  yargs
    .strict(false) // so that we can forward arguments to jest
    .positional('filter', {
      default: getProject().sides,
      description:
        'Which side(s) to test, and/or a regular expression to match against your test files to filter by',
      type: 'array',
    })
    .option('watch', {
      describe:
        'Run tests related to changed files based on hg/git. Specify the name or path to a file to focus on a specific set of tests',
      type: 'boolean',
      default: true,
    })
    .option('collect-coverage', {
      describe:
        'Show test coverage summary and output info to coverage directory',
      type: 'boolean',
      default: false,
    })
    .option('db-push', {
      describe:
        "Syncs the test database with your Prisma schema without requiring a migration. It creates a test database if it doesn't already exist.",
      type: 'boolean',
      default: true,
    })
    .epilogue(
      `Also see the ${terminalLink(
        'Redwood CLI Reference',
        'https://redwoodjs.com/reference/command-line-interface#test'
      )}`
    )
}

export const handler = async ({
  filter: filterParams = [],
  watch = true,
  collectCoverage = false,
  dbPush = true,
  ...others
}) => {
  const rwjsPaths = getPaths()
  const forwardJestFlags = Object.keys(others).flatMap((flagName) => {
    if (
      ['watch', 'collect-coverage', 'db-push', '$0', '_'].includes(flagName)
    ) {
      // filter out flags meant for the rw test command only
      return []
    } else {
      // and forward on the other flags
      return [
        flagName.length > 1 ? `--${flagName}` : `-${flagName}`,
        others[flagName],
      ]
    }
  })

  // Only the side params
  const sides = filterParams.filter((filterString) =>
    getProject().sides.includes(filterString)
  )

  // All the other params, apart from sides
  const jestFilterArgs = [
    ...filterParams.filter(
      (filterString) => !getProject().sides.includes(filterString)
    ),
  ]

  const jestArgs = [
    ...jestFilterArgs,
    ...forwardJestFlags,
    collectCoverage ? '--collectCoverage' : null,
    '--passWithNoTests',
    ...jestFilterArgs,
    sides.includes('api') && '--runInBand',
  ].filter((flagOrValue) => flagOrValue !== null) // Filter out nulls, not booleans because user may have passed a --something false flag

  // If the user wants to watch, set the proper watch flag based on what kind of repo this is
  // because of https://github.com/facebook/create-react-app/issues/5210
  if (watch && !process.env.CI && !collectCoverage) {
    const hasSourceControl = isInGitRepository() || isInMercurialRepository()
    jestArgs.push(hasSourceControl ? '--watch' : '--watchAll')
  }

  // if no sides declared with yargs, default to all sides
  if (!sides.length) {
    getProject().sides.forEach((side) => sides.push(side))
  }

  jestArgs.push(
    '--config',
    `"${require.resolve('@redwoodjs/testing/config/jest/jest.config.js')}"`
  )

  if (sides.length > 0) {
    jestArgs.push('--projects', ...sides)
  }

  try {
    const cacheDirDb = `file:${ensurePosixPath(
      rwjsPaths.generated.base
    )}/test.db`
    const DATABASE_URL = process.env.TEST_DATABASE_URL || cacheDirDb

    if (sides.includes('api') && dbPush) {
      // Sync||create test database
      await execa(
        `yarn rw`,
        ['prisma db push', '--force-reset', '--accept-data-loss'],
        {
          cwd: rwjsPaths.api.base,
          stdio: 'inherit',
          shell: true,
          env: { DATABASE_URL },
        }
      )
    }

    // **NOTE** There is no official way to run Jest programmatically,
    // so we're running it via execa, since `jest.run()` is a bit unstable.
    // https://github.com/facebook/jest/issues/5048
    await execa('yarn jest', jestArgs, {
      cwd: rwjsPaths.base,
      shell: true,
      stdio: 'inherit',
      env: { DATABASE_URL },
    })
  } catch (e) {
    // Errors already shown from execa inherited stderr
    process.exit(e?.exitCode || 1)
  }
}
