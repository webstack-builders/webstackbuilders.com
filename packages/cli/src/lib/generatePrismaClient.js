// helper used in Dev and Build commands

import fs from 'fs'
import path from 'path'

import { runCommandTask, getPaths } from '../lib'

const skipTask = (schema = getPaths().api.dbSchema) => {
  if (!fs.existsSync(schema)) {
    console.log(
      `Skipping database and Prisma client generation, no \`schema.prisma\` file found: \`${schema}\``
    )
    return true
  }
  return false
}

export const generatePrismaCommand = (schema) => {
  if (skipTask(schema)) {
    return {}
  }

  return {
    cmd: 'yarn prisma',
    args: ['generate', schema && `--schema="${schema}"`],
  }
}

/**
 * Conditionally generate the prisma client, skip if it already exists.
 */
export const generatePrismaClient = async ({
  verbose = true,
  force = true,
  schema = getPaths().api.dbSchema,
}) => {
  if (skipTask(schema)) {
    return
  }

  // Do not generate the Prisma client if it exists.
  if (!force) {
    // The Prisma client throws if it is not generated.
    try {
      // Import the client from the redwood apps node_modules path.
      const { PrismaClient } = require(path.join(
        getPaths().base,
        'node_modules/.prisma/client'
      ))
      // eslint-disable-next-line
      new PrismaClient()
      return // Client exists, so abort.
    } catch (e) {
      // Swallow your pain, and generate.
    }
  }

  return await runCommandTask(
    [
      {
        title: 'Generating the Prisma client...',
        ...generatePrismaCommand(schema),
      },
    ],
    {
      verbose,
    }
  )
}
