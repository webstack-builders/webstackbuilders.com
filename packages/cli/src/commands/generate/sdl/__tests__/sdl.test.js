global.__dirname = __dirname
import path from 'path'

// Load mocks
import '../../../../lib/test'

import { getDefaultArgs } from '../../../../lib'
import * as sdl from '../sdl'

afterEach(() => {
  jest.clearAllMocks()
})

const extensionForBaseArgs = (baseArgs) =>
  baseArgs && baseArgs.typescript ? 'ts' : 'js'

const itReturnsExactlyFourFiles = (baseArgs = {}) => {
  test('returns exactly 4 files', async () => {
    const files = await sdl.files({ ...baseArgs, name: 'Post', crud: false })

    expect(Object.keys(files).length).toEqual(4)
  })
}

// in this case we'll trust that a service and test are actually created
// with the correct filename, but the contents of that file should be the
// job of the service tests
const itCreatesAService = (baseArgs = {}) => {
  test('creates a service', async () => {
    const files = await sdl.files({ ...baseArgs, name: 'User', crud: false })
    const extension = extensionForBaseArgs(baseArgs)

    expect(files).toHaveProperty([
      path.normalize(
        `/path/to/project/api/src/services/users/users.${extension}`
      ),
    ])
    expect(files).toHaveProperty([
      path.normalize(
        `/path/to/project/api/src/services/users/users.test.${extension}`
      ),
    ])
  })
}

const itCreatesASingleWordSDLFile = (baseArgs = {}) => {
  test('creates a single word sdl file', async () => {
    const files = await sdl.files({ ...baseArgs, name: 'User' })
    const extension = extensionForBaseArgs(baseArgs)

    expect(
      files[
        path.normalize(
          `/path/to/project/api/src/graphql/users.sdl.${extension}`
        )
      ]
    ).toMatchSnapshot()
  })
}

const itCreatesAMultiWordSDLFile = (baseArgs = {}) => {
  test('creates a multi word sdl file', async () => {
    const files = await sdl.files({
      ...baseArgs,
      name: 'UserProfile',
    })
    const extension = extensionForBaseArgs(baseArgs)

    expect(
      files[
        path.normalize(
          `/path/to/project/api/src/graphql/userProfiles.sdl.${extension}`
        )
      ]
    ).toMatchSnapshot()
  })
}

const itCreatesASingleWordSDLFileWithCRUD = (baseArgs = {}) => {
  test('creates a single word sdl file with CRUD actions', async () => {
    const files = await sdl.files({ ...baseArgs, name: 'Post', crud: true })
    const extension = extensionForBaseArgs(baseArgs)

    expect(
      files[
        path.normalize(
          `/path/to/project/api/src/graphql/posts.sdl.${extension}`
        )
      ]
    ).toMatchSnapshot()
  })
}

const itCreateAMultiWordSDLFileWithCRUD = (baseArgs = {}) => {
  test('creates a multi word sdl file with CRUD actions', async () => {
    const files = await sdl.files({
      ...baseArgs,
      name: 'UserProfile',
      crud: true,
    })
    const extension = extensionForBaseArgs(baseArgs)

    expect(
      files[
        path.normalize(
          `/path/to/project/api/src/graphql/userProfiles.sdl.${extension}`
        )
      ]
    ).toMatchSnapshot()
  })
}

const itCreatesAnSDLFileWithEnumDefinitions = (baseArgs = {}) => {
  test('creates a sdl file with enum definitions', async () => {
    const files = await sdl.files({ ...baseArgs, name: 'Shoe', crud: true })
    const extension = extensionForBaseArgs(baseArgs)

    expect(
      files[
        path.normalize(
          `/path/to/project/api/src/graphql/shoes.sdl.${extension}`
        )
      ]
    ).toMatchSnapshot()
  })
}

const itCreatesAnSDLFileWithJsonDefinitions = (baseArgs = {}) => {
  test('creates a sdl file with json definitions', async () => {
    const files = await sdl.files({ ...baseArgs, name: 'Photo', crud: true })
    const extension = extensionForBaseArgs(baseArgs)

    expect(
      files[
        path.normalize(
          `/path/to/project/api/src/graphql/photos.sdl.${extension}`
        )
      ]
    ).toMatchSnapshot()
  })
}

describe('in javascript mode', () => {
  const baseArgs = { ...getDefaultArgs(sdl.defaults), tests: true }

  itReturnsExactlyFourFiles(baseArgs)
  itCreatesAService(baseArgs)
  itCreatesASingleWordSDLFile(baseArgs)
  itCreatesAMultiWordSDLFile(baseArgs)
  itCreatesASingleWordSDLFileWithCRUD(baseArgs)
  itCreateAMultiWordSDLFileWithCRUD(baseArgs)
  itCreatesAnSDLFileWithEnumDefinitions(baseArgs)
  itCreatesAnSDLFileWithJsonDefinitions(baseArgs)
})

describe('in typescript mode', () => {
  const baseArgs = {
    ...getDefaultArgs(sdl.defaults),
    typescript: true,
    tests: true,
  }

  itReturnsExactlyFourFiles(baseArgs)
  itCreatesAService(baseArgs)
  itCreatesASingleWordSDLFile(baseArgs)
  itCreatesAMultiWordSDLFile(baseArgs)
  itCreatesASingleWordSDLFileWithCRUD(baseArgs)
  itCreateAMultiWordSDLFileWithCRUD(baseArgs)
  itCreatesAnSDLFileWithEnumDefinitions(baseArgs)
  itCreatesAnSDLFileWithJsonDefinitions(baseArgs)
})
