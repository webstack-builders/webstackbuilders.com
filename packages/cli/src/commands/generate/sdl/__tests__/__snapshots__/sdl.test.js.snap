// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`in javascript mode creates a multi word sdl file 1`] = `
"export const schema = gql\`
  type UserProfile {
    id: Int!
    username: String!
    userId: Int!
    user: User!
  }

  type Query {
    userProfiles: [UserProfile!]!
  }

  input CreateUserProfileInput {
    username: String!
    userId: Int!
  }

  input UpdateUserProfileInput {
    username: String
    userId: Int
  }
\`
"
`;

exports[`in javascript mode creates a multi word sdl file with CRUD actions 1`] = `
"export const schema = gql\`
  type UserProfile {
    id: Int!
    username: String!
    userId: Int!
    user: User!
  }

  type Query {
    userProfiles: [UserProfile!]!
    userProfile(id: Int!): UserProfile
  }

  input CreateUserProfileInput {
    username: String!
    userId: Int!
  }

  input UpdateUserProfileInput {
    username: String
    userId: Int
  }

  type Mutation {
    createUserProfile(input: CreateUserProfileInput!): UserProfile!
    updateUserProfile(id: Int!, input: UpdateUserProfileInput!): UserProfile!
    deleteUserProfile(id: Int!): UserProfile!
  }
\`
"
`;

exports[`in javascript mode creates a sdl file with enum definitions 1`] = `
"export const schema = gql\`
  type Shoe {
    id: Int!
    color: Color!
  }

  enum Color {
    RED
    GREEN
    BLUE
  }

  type Query {
    shoes: [Shoe!]!
    shoe(id: Int!): Shoe
  }

  input CreateShoeInput {
    color: Color!
  }

  input UpdateShoeInput {
    color: Color
  }

  type Mutation {
    createShoe(input: CreateShoeInput!): Shoe!
    updateShoe(id: Int!, input: UpdateShoeInput!): Shoe!
    deleteShoe(id: Int!): Shoe!
  }
\`
"
`;

exports[`in javascript mode creates a sdl file with json definitions 1`] = `
"export const schema = gql\`
  type Photo {
    id: Int!
    name: String!
    metadata: JSON!
  }

  type Query {
    photos: [Photo!]!
    photo(id: Int!): Photo
  }

  input CreatePhotoInput {
    name: String!
    metadata: JSON!
  }

  input UpdatePhotoInput {
    name: String
    metadata: JSON
  }

  type Mutation {
    createPhoto(input: CreatePhotoInput!): Photo!
    updatePhoto(id: Int!, input: UpdatePhotoInput!): Photo!
    deletePhoto(id: Int!): Photo!
  }
\`
"
`;

exports[`in javascript mode creates a single word sdl file 1`] = `
"export const schema = gql\`
  type User {
    id: Int!
    name: String
    email: String!
    isAdmin: Boolean!
    profiles: [UserProfile]!
  }

  type Query {
    users: [User!]!
  }

  input CreateUserInput {
    name: String
    email: String!
    isAdmin: Boolean!
  }

  input UpdateUserInput {
    name: String
    email: String
    isAdmin: Boolean
  }
\`
"
`;

exports[`in javascript mode creates a single word sdl file with CRUD actions 1`] = `
"export const schema = gql\`
  type Post {
    id: Int!
    title: String!
    slug: String!
    author: String!
    body: String!
    image: String
    postedAt: DateTime
  }

  type Query {
    posts: [Post!]!
    post(id: Int!): Post
  }

  input CreatePostInput {
    title: String!
    slug: String!
    author: String!
    body: String!
    image: String
    postedAt: DateTime
  }

  input UpdatePostInput {
    title: String
    slug: String
    author: String
    body: String
    image: String
    postedAt: DateTime
  }

  type Mutation {
    createPost(input: CreatePostInput!): Post!
    updatePost(id: Int!, input: UpdatePostInput!): Post!
    deletePost(id: Int!): Post!
  }
\`
"
`;

exports[`in typescript mode creates a multi word sdl file 1`] = `
"export const schema = gql\`
  type UserProfile {
    id: Int!
    username: String!
    userId: Int!
    user: User!
  }

  type Query {
    userProfiles: [UserProfile!]!
  }

  input CreateUserProfileInput {
    username: String!
    userId: Int!
  }

  input UpdateUserProfileInput {
    username: String
    userId: Int
  }
\`
"
`;

exports[`in typescript mode creates a multi word sdl file with CRUD actions 1`] = `
"export const schema = gql\`
  type UserProfile {
    id: Int!
    username: String!
    userId: Int!
    user: User!
  }

  type Query {
    userProfiles: [UserProfile!]!
    userProfile(id: Int!): UserProfile
  }

  input CreateUserProfileInput {
    username: String!
    userId: Int!
  }

  input UpdateUserProfileInput {
    username: String
    userId: Int
  }

  type Mutation {
    createUserProfile(input: CreateUserProfileInput!): UserProfile!
    updateUserProfile(id: Int!, input: UpdateUserProfileInput!): UserProfile!
    deleteUserProfile(id: Int!): UserProfile!
  }
\`
"
`;

exports[`in typescript mode creates a sdl file with enum definitions 1`] = `
"export const schema = gql\`
  type Shoe {
    id: Int!
    color: Color!
  }

  enum Color {
    RED
    GREEN
    BLUE
  }

  type Query {
    shoes: [Shoe!]!
    shoe(id: Int!): Shoe
  }

  input CreateShoeInput {
    color: Color!
  }

  input UpdateShoeInput {
    color: Color
  }

  type Mutation {
    createShoe(input: CreateShoeInput!): Shoe!
    updateShoe(id: Int!, input: UpdateShoeInput!): Shoe!
    deleteShoe(id: Int!): Shoe!
  }
\`
"
`;

exports[`in typescript mode creates a sdl file with json definitions 1`] = `
"export const schema = gql\`
  type Photo {
    id: Int!
    name: String!
    metadata: JSON!
  }

  type Query {
    photos: [Photo!]!
    photo(id: Int!): Photo
  }

  input CreatePhotoInput {
    name: String!
    metadata: JSON!
  }

  input UpdatePhotoInput {
    name: String
    metadata: JSON
  }

  type Mutation {
    createPhoto(input: CreatePhotoInput!): Photo!
    updatePhoto(id: Int!, input: UpdatePhotoInput!): Photo!
    deletePhoto(id: Int!): Photo!
  }
\`
"
`;

exports[`in typescript mode creates a single word sdl file 1`] = `
"export const schema = gql\`
  type User {
    id: Int!
    name: String
    email: String!
    isAdmin: Boolean!
    profiles: [UserProfile]!
  }

  type Query {
    users: [User!]!
  }

  input CreateUserInput {
    name: String
    email: String!
    isAdmin: Boolean!
  }

  input UpdateUserInput {
    name: String
    email: String
    isAdmin: Boolean
  }
\`
"
`;

exports[`in typescript mode creates a single word sdl file with CRUD actions 1`] = `
"export const schema = gql\`
  type Post {
    id: Int!
    title: String!
    slug: String!
    author: String!
    body: String!
    image: String
    postedAt: DateTime
  }

  type Query {
    posts: [Post!]!
    post(id: Int!): Post
  }

  input CreatePostInput {
    title: String!
    slug: String!
    author: String!
    body: String!
    image: String
    postedAt: DateTime
  }

  input UpdatePostInput {
    title: String
    slug: String
    author: String
    body: String
    image: String
    postedAt: DateTime
  }

  type Mutation {
    createPost(input: CreatePostInput!): Post!
    updatePost(id: Int!, input: UpdatePostInput!): Post!
    deletePost(id: Int!): Post!
  }
\`
"
`;
